From 3855ab79b43c1195bd6a78764fbc249fc62ec236 Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Sun, 16 Mar 2014 18:57:58 +0100
Subject: [PATCH 1/2] _oasis: make it possible to not build tests, docs and lwt
 support.

---
 _oasis | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/_oasis b/_oasis
index c17a6b7..7fc2f6f 100644
--- a/_oasis
+++ b/_oasis
@@ -20,6 +20,10 @@ Description:
   .
   [libarchive website](http://code.google.com/p/libarchive/)
 
+Flag "archivelwt"
+  Description: Also provide operations as lwt threads
+  Default:     true
+
 Library "archive-ocaml"
   Path:       src
   Modules:    Archive, ArchiveLow
@@ -29,6 +33,7 @@ Library "archive-ocaml"
   BuildDepends: unix, fileutils
 
 Library "archive-lwt"
+  Build$:        flag(archivelwt)
   Path:          src
   BuildDepends:  archive, lwt (>= 2.3.2), lwt.preemptive, threads
   Modules:       ArchiveLwt
@@ -36,6 +41,7 @@ Library "archive-lwt"
   FindlibName:   lwt
 
 Executable test
+  Build$:       flag(tests)
   Path:         test
   BuildTools:   ocamlbuild
   MainIs:       test.ml
@@ -47,6 +53,7 @@ Test main
   Command: $test
 
 Document "api-archive"
+  Build$: flag(docs)
   Title:      API reference for Archive
   Type:       ocamlbuild (0.2)
   InstallDir: $htmldir/api-archive
-- 
1.8.4

